Compiler warning on RTEMS

**Describe the bug**
When compiling on RTEMS this application now has multiple warnings:

```
/home/joe/code/cfecfs/github/apps/sample_app/fsw/src/sample_app.c: In function 'SAMPLE_AppInit':
/home/joe/code/cfecfs/github/apps/sample_app/fsw/src/sample_app.c:166:30: error: format '%X' expects argument of type 'unsigned int', but argument 2 has type 'int32' [-Werror=format=]
                              status);
                              ^
/home/joe/code/cfecfs/github/apps/sample_app/fsw/src/sample_app.c:187:30: error: format '%X' expects argument of type 'unsigned int', but argument 2 has type 'int32' [-Werror=format=]
                              status);
                              ^
/home/joe/code/cfecfs/github/apps/sample_app/fsw/src/sample_app.c:199:30: error: format '%X' expects argument of type 'unsigned int', but argument 2 has type 'int32' [-Werror=format=]
                              status);
                              ^
/home/joe/code/cfecfs/github/apps/sample_app/fsw/src/sample_app.c:211:30: error: format '%X' expects argument of type 'unsigned int', but argument 2 has type 'int32' [-Werror=format=]
                              status);
                              ^
/home/joe/code/cfecfs/github/apps/sample_app/fsw/src/sample_app.c:226:9: error: format '%X' expects argument of type 'unsigned int', but argument 2 has type 'int32' [-Werror=format=]
         CFE_ES_WriteToSysLog("Sample App: Error Registering \
         ^
/home/joe/code/cfecfs/github/apps/sample_app/fsw/src/sample_app.c: In function 'SAMPLE_GetCrc':
/home/joe/code/cfecfs/github/apps/sample_app/fsw/src/sample_app.c:535:9: error: format '%X' expects argument of type 'unsigned int', but argument 2 has type 'uint32' [-Werror=format=]
         CFE_ES_WriteToSysLog("Sample App: CRC: 0x%08X\n\n", Crc);
         ^
```

**To Reproduce**
Build on RTEMS per the readme/howto and with `-Wall -Werror` switches

**Expected behavior**
Should build clean

**System observed on:**
Ubuntu 18.04 (build host) when cross compiling for RTEMS 4.11

**Additional context**
Appears the problem calls were introduced in #1 

As a rule of thumb, whenever "printf" style conversions are used in conjunction with the fixed-width types (int32, uint32, etc) these need an explicit cast because the actual type definitions vary by platform.

**Reporter Info**
Joseph Hickey, Vantage Systems, Inc.

--- a/fsw/src/sample_app.c
+++ b/fsw/src/sample_app.c
@@ -163,8 +163,8 @@ int32 SAMPLE_AppInit( void )
                               CFE_EVS_BINARY_FILTER);
     if (status != CFE_SUCCESS)
     {
-        CFE_ES_WriteToSysLog("Sample App: Error Registering Events, RC = 0x%08X\n",
-                             status);
+        CFE_ES_WriteToSysLog("Sample App: Error Registering Events, RC = 0x%08lX\n",
+                             (unsigned long)status);
         return ( status );
     }
 
@@ -184,8 +184,8 @@ int32 SAMPLE_AppInit( void )
                                Sample_AppData.PipeName);
     if (status != CFE_SUCCESS)
     {
-        CFE_ES_WriteToSysLog("Sample App: Error creating pipe, RC = 0x%08X\n",
-                             status);
+        CFE_ES_WriteToSysLog("Sample App: Error creating pipe, RC = 0x%08lX\n",
+                             (unsigned long)status);
         return ( status );
     }
 
@@ -196,8 +196,8 @@ int32 SAMPLE_AppInit( void )
                               Sample_AppData.SAMPLE_CommandPipe);
     if (status != CFE_SUCCESS)
     {
-        CFE_ES_WriteToSysLog("Sample App: Error Subscribing to HK request, RC = 0x%08X\n", 
-                             status);
+        CFE_ES_WriteToSysLog("Sample App: Error Subscribing to HK request, RC = 0x%08lX\n",
+                             (unsigned long)status);
         return ( status );
     }
 
@@ -208,8 +208,8 @@ int32 SAMPLE_AppInit( void )
                               Sample_AppData.SAMPLE_CommandPipe);
     if (status != CFE_SUCCESS )
     {
-        CFE_ES_WriteToSysLog("Sample App: Error Subscribing to Command, RC = 0x%08X\n",
-                             status);
+        CFE_ES_WriteToSysLog("Sample App: Error Subscribing to Command, RC = 0x%08lX\n",
+                             (unsigned long)status);
 
         return ( status );
     }
@@ -225,7 +225,7 @@ int32 SAMPLE_AppInit( void )
     if ( status != CFE_SUCCESS )
     {
         CFE_ES_WriteToSysLog("Sample App: Error Registering \
-                              Table, RC = 0x%08X\n", status);
+                              Table, RC = 0x%08lX\n", (unsigned long)status);
 
         return ( status );
     }
@@ -524,7 +524,7 @@ void SAMPLE_GetCrc( const char *TableName )
     else
     {
         Crc = TblInfoPtr.Crc;
-        CFE_ES_WriteToSysLog("Sample App: CRC: 0x%08X\n\n", Crc);
+        CFE_ES_WriteToSysLog("Sample App: CRC: 0x%08lX\n\n", (unsigned long)Crc);
     }
 
     return;

